src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -89,10 +93,17 @@ public CommandResult execute(Model model) throws CommandException {
             model.addPersonKeepFilter(personToEdit);
             throw new CommandException(MESSAGE_DUPLICATE_FIELD);
         }
-
         model.addPersonKeepFilter(editedPerson);
-        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
-        return new CommandResult(String.format(MESSAGE_EDIT_PERSON_SUCCESS, Messages.format(editedPerson)));
+    }
+
+    private void updateLastViewedPersonIfNecessary(Person personToEdit, Person editedPerson, Model model) {
+        Optional<Person> lastViewedPerson = model.getLastViewedPerson();
+        if (lastViewedPerson.isEmpty()) {
+            return;
+        }
+        if (lastViewedPerson.get().equals(personToEdit)) {
+            model.updateLastViewedPerson(editedPerson);
+        }
```

Comment:
Likewise, the code could be more elegant here.

Reply:
```
private void updateLastViewedPersonIfNecessary(Person personToEdit, Person editedPerson, Model model) {
    model.getLastViewedPerson()
         .filter(lastViewedPerson -> lastViewedPerson.equals(personToEdit))
         .ifPresent(lastViewedPerson -> model.updateLastViewedPerson(editedPerson));
}
```

Reply:
Thanks!